code/game/objects/items/weapons/defib.dm

```diff
@@ -217,35 +214,79 @@
 	desc = "A belt-equipped defibrillator that can be rapidly deployed."
 	icon_state = "defibcompact"
 	item_state = "defibcompact"
+	sprite_sheets = null //Because Vox had the belt defibrillator sprites in back.dm
 	w_class = WEIGHT_CLASS_NORMAL
 	slot_flags = SLOT_BELT
 	origin_tech = "biotech=5"
 
+/obj/item/defibrillator/compact/loaded/Initialize(mapload)
+	. = ..()
+	cell = new(src)
+	update_icon()
+
 /obj/item/defibrillator/compact/item_action_slot_check(slot, mob/user)
 	if(slot == slot_belt)
 		return TRUE
 
-/obj/item/defibrillator/compact/loaded/New()
-	..()
-	paddles = make_paddles()
-	cell = new(src)
-	update_icon()
-	return
-
 /obj/item/defibrillator/compact/combat
 	name = "combat defibrillator"
 	desc = "A belt-equipped blood-red defibrillator that can be rapidly deployed. Does not have the restrictions or safeties of conventional defibrillators and can revive through space suits."
+	icon_state = "defibcombat"
+	item_state = "defibcombat"
+	paddle_type = /obj/item/twohanded/shockpaddles/syndicate
 	combat = TRUE
 	safety = FALSE
 
-/obj/item/defibrillator/compact/combat/loaded/New()
-	..()
-	paddles = make_paddles()
+/obj/item/defibrillator/compact/combat/loaded/Initialize(mapload)
+	. = ..()
 	cell = new(src)
 	update_icon()
-	return
 
-/obj/item/defibrillator/compact/combat/attackby(obj/item/W, mob/user, params)
+/obj/item/defibrillator/compact/advanced
+	name = "advanced compact defibrillator"
+	desc = "A belt-equipped state-of-the-art defibrillator that can be rapidly deployed in all enviroments. Uses an experimental self-charging cell, meaning that it will (probably) never stop working. Can revive through space suits. Impossible to hack or damage."
+	icon_state = "defibadvanced"
+	item_state = "defibadvanced"
+	paddle_type = /obj/item/twohanded/shockpaddles/advanced
+	combat = TRUE
+	safety = TRUE
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF //Objective item, better not have it destroyed.
+
+	var/next_emag_message //to prevent spam from the emagging message on the advanced defibrillator
+	var/next_emp_message //same as above, but for EMPs
+
+/obj/item/defibrillator/compact/advanced/attackby(obj/item/W, mob/user, params)
+	if(W == paddles)
+		paddles.unwield()
+		toggle_paddles()
+		update_icon()
+		return
+
+/obj/item/defibrillator/compact/advanced/loaded/Initialize(mapload)
+	. = ..()
+	cell = new /obj/item/stock_parts/cell/bluespace/charging(src)
+	update_icon()
+
+/obj/item/defibrillator/compact/advanced/emag_act(mob/user)
+	if(world.time > next_emag_message)
+		to_chat(user, "<span class='warning'>You try to silently disable [src]'s safety protocols with the card but it rejects your attempt... uh oh...")
```

Comment:
```suggestion
		to_chat(user, "<span class='warning'>You try to silently disable [src]'s safety protocols with the sequencer, but it rejects your attempt. Uh oh...")
```
Card? Sequencer? Maybe its not that important but I'll suggest it.

Reply:
I think just "card" is descriptive enough, at least to me.